frontend/packages/citizen-frontend/src/messages/MessagesPage.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react'
+import Container from '@evaka/lib-components/src/layout/Container'
+import { Gap } from '@evaka/lib-components/src/white-space'
+import MessagesList from '~messages/MessagesList'
+import MessageReadView from '~messages/MessageReadView'
+import { Loading, Result, Success } from '@evaka/lib-common/src/api'
+import { ReceivedBulletin } from '~messages/types'
+import { useRestApi } from '@evaka/lib-common/src/utils/useRestApi'
+import { getBulletins, markBulletinRead } from '~messages/api'
+import AdaptiveFlex from '@evaka/lib-components/src/layout/AdaptiveFlex'
+import { messagesBreakpoint } from '~messages/const'
+
+function MessagesPage() {
+  const [bulletins, setBulletins] = useState<Result<ReceivedBulletin[]>>(
+    Loading.of()
+  )
+  const [activeBulletin, setActiveBulletin] = useState<ReceivedBulletin | null>(
+    null
+  )
+
+  const loadBulletins = useRestApi(getBulletins, setBulletins)
+  useEffect(() => loadBulletins(), [])
+
+  const openBulletin = (bulletin: ReceivedBulletin) => {
+    markBulletinRead(bulletin.id)
+
+    setActiveBulletin({
+      ...bulletin,
+      isRead: true
+    })
+
+    if (bulletins.isSuccess) {
+      setBulletins(
+        Success.of(
+          bulletins.value.map((b) =>
+            b.id === bulletin.id ? { ...b, isRead: true } : b
+          )
+        )
```

Comment:
You can just do `bulletins.map((b) => ...)` here.

Reply:
computer says no
![Screenshot 2021-02-24 at 13 08 07](https://user-images.githubusercontent.com/1533854/108991926-5d2d6380-76a1-11eb-8217-24497fabce86.png)

Reply:
after map b is still an array

Reply:
Yeah sorry I had a bit of a brain fart, but what I meant is that you can remove the whole `bulletins.isSuccess` check and manual `Success.of(...)` wrapping and just map `bulletins.`